diff --git a/fizz/experimental/ktls/LinuxKTLS.h b/fizz/experimental/ktls/LinuxKTLS.h
index ba6a95f..6ca3f5c 100644
--- a/fizz/experimental/ktls/LinuxKTLS.h
+++ b/fizz/experimental/ktls/LinuxKTLS.h
@@ -106,4 +106,16 @@
 #define TLS_CIPHER_AES_GCM_256_REC_SEQ_SIZE 8
 #endif
 
+#ifndef TCP_ULP
+#define TCP_ULP 31
+#endif
+
+#ifndef SOL_TCP
+#define SOL_TCP 6
+#endif
+
+#ifndef SOL_TLS
+#define SOL_TLS 282
+#endif
+
 #endif
