diff --git a/third_party/fsst/libfsst.hpp b/third_party/fsst/libfsst.hpp
--- a/third_party/fsst/libfsst.hpp
+++ b/third_party/fsst/libfsst.hpp
@@ -17,6 +17,7 @@
 // You can contact the authors via the FSST source repository : https://github.com/cwida/fsst 
 #include <algorithm>
 #include <cassert>
+#include <cstdint>
 #include <cstring>
 #include <fstream>
 #include <iostream>
diff --git a/third_party/thrift/thrift/transport/TTransport.h b/third_party/thrift/thrift/transport/TTransport.h
--- a/third_party/thrift/thrift/transport/TTransport.h
+++ b/third_party/thrift/thrift/transport/TTransport.h
@@ -23,6 +23,7 @@
 #include "thrift/Thrift.h"
 #include "thrift/transport/TTransportException.h"
 #include <memory>
+#include <cstdint>
 #include <string>
 
 namespace duckdb_apache {
