diff --git a/gen_uuid.c b/gen_uuid.c
--- a/gen_uuid.c
+++ b/gen_uuid.c
@@ -35,8 +35,15 @@
 /*
  * Force inclusion of SVID stuff since we need it if we're compiling in
  * gcc-wall wall mode
+ * Enable BSD extensions for usleep, using both old and new glibc macros.
  */
 #define _SVID_SOURCE
+#ifndef _BSD_SOURCE
+#define _BSD_SOURCE
+#endif
+#ifndef _DEFAULT_SOURCE
+#define _DEFAULT_SOURCE
+#endif
 
 #ifdef _WIN32
 #define _WIN32_WINNT 0x0500
